### ISSUE TITLE
Add Markdown syntax guide

### ISSUE THREAD

**Issue by justinkelly** [OscarGodson/EpicEditor#148]
1 feature that would be useful is to add a markdown reference button in the UI as an icon

so if someone doesnt know markdown they can click the button and it provides a summary of the mardown syntax

Cheers

Justin

**Comment by OscarGodson** [comment]
This would probably be an extension. EpicEditor is moving away from purely being an Markdown editor. It'll always have Markdown as the default parser, but it's meant to be any kind of editor you want. So, we'll probably end up making a MD guide as a EpicEditor package. 

Great idea tho! i'll add it to the list of extensions we should make after we build out the extension stuff. :)

**Comment by justinkelly** [comment]
thanks oscar for the reply - great work on the editor - hopefully this gets implemented as an extendible feature in the future
- so if you using epic for rst you would get a rst guide etc..

**Comment by OscarGodson** [comment]
Yep! we'll probably provide hooks into a few things. Now that you mentioned that it'd be a great idea if packages could include syntax guides which are all formatted and styled the same, you just need to write it. Thanks!

**Comment by Le-zz-ge-zz-n-zz-d** [comment]
@OscarGodson: Could you please clarify what you mean when you say "EpicEditor is moving away from purely being a Markdown editor"? What is moving towards? Sorry if this comes from nowhere - I saw this on HN today and was checking it out.

**Comment by OscarGodson** [comment]
@Legend It's moving towards a general purpose editor that's easily extendible by developers via it's API's and extension API's but also a really slick editor for the users. We don't want to require developers or users to use markdown to use the editor. For example, wikipedia could put it in and swap it out by just including http://remysharp.com/downloads/wiki2html.js and then doing:

``` js
new EpicEditor({
  parser: function (str) {
    return str.wiki2html();
  }
});
```

We just don't want to get too coupled to Markdown. We want all basic feature to work for all parsers.

**Comment by Le-zz-ge-zz-n-zz-d** [comment]
@OscarGodson That makes a lot of sense. Thank you for the clarification and of course, your wonderful work!

**Comment by bewildergeist** [comment]
In terms of layout and styling, perhaps this feature could borrow a few ideas from Github's Gollum-editor:

https://github.com/gollum/gollum/wiki/Home/_edit

(Click the help/question mark icon to fold out the syntax help, and not how its content changes as you change "Edit mode" from the dropdown).

![gollum_editor](https://f.cloud.github.com/assets/121186/566963/1abbd73e-c6a5-11e2-8d94-95219432a5c3.png)
